### ISSUE TITLE
ERROR at setup of reshape_image_to_4d_superpixel with numpy 1.11 beta 2

### ISSUE THREAD

**Issue by olebole** [sunpy/sunpy#1653]
With the new beta release of numpy (which just entered Debian unstable), I get the following test setup error:

```
__________________ ERROR at setup of test_all_maps_same_shape __________________

    @pytest.fixture()
    def mapcube_different():
        testpath = sunpy.data.test.rootdir
        aia_file = os.path.join(testpath, "aia_171_level1.fits")
        aia_map = sunpy.map.Map(aia_file)
>       return sunpy.map.Map([aia_map, aia_map.superpixel((4, 4)*u.pix)], cube=True)

/usr/lib/python2.7/dist-packages/sunpy/map/tests/test_mapcube.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python2.7/dist-packages/astropy/utils/decorators.py:679: in superpixel
    func = make_function_with_signature(func, name=name, **wrapped_args)
/usr/lib/python2.7/dist-packages/astropy/units/decorators.py:130: in wrapper
    return wrapped_function(*func_args, **func_kwargs)
/usr/lib/python2.7/dist-packages/sunpy/map/mapbase.py:1081: in superpixel
    [dimensions.value[1], dimensions.value[0]])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

img = array([[-1.25,  0.  ,  1.  , ...,  0.  ,  0.5 , -0.75],
       [ 0.75, -0.25, ...  0.75, -0.75,  0.  ],
       [ 0.75,  1.5 , -0.75, ...,  0.  , -0.5 ,  0.5 ]])
dimensions = [4.0, 4.0]

    def reshape_image_to_4d_superpixel(img, dimensions):
[...]
        return img.reshape(img.shape[0] / dimensions[0],
                           dimensions[0],
                           img.shape[1] / dimensions[1],
>                          dimensions[1])
E       TypeError: 'numpy.float64' object cannot be interpreted as an index

/usr/lib/python2.7/dist-packages/sunpy/image/rescale.py:194: TypeError
```

A similar error, I get for `test_as_array`.
Full test log [here](https://ci.debian.net/data/packages/unstable/amd64/s/sunpy/20160131_053403.autopkgtest.log.gz)

**Comment by olebole** [comment]
I also get a number of failures that are caused by invalid slices. One example:
```
_______________________________ test_scale[0.25] _______________________________

scale_factor = 0.25

    @pytest.mark.parametrize("scale_factor", [0.25, 0.5, 0.75, 1.0, 1.25, 1.5])
    def test_scale(scale_factor):
        # No rotation for all scaling tests.
        rmatrix = np.array([[1.0, 0.0], [0.0, 1.0]])
    
        # Check a scaled image against the expected outcome
        newim = tf.rescale(original/original.max(), scale_factor, order=4,
                           mode='constant') * original.max()
        # Old width and new center of image
        w = original.shape[0]/2.0 - 0.5
        new_c = (newim.shape[0]/2.0) - 0.5
        expected = np.zeros(original.shape)
        upper = w+new_c+1
        if scale_factor > 1:
            lower = new_c-w
            expected = newim[lower:upper, lower:upper]
        else:
            lower = w-new_c
>           expected[lower:upper, lower:upper] = newim
E           IndexError: invalid slice

/usr/lib/python2.7/dist-packages/sunpy/image/tests/test_transform.py:122: IndexError
```
This happens with `test_clip_edges`, `test_calculate_match_template_shift`, `test_mapcube_coalign_by_match_template`, `test_apply_shifts`, many `test_scale` parameters, many `test_all` parameters, `test_submap`, `test_rotate`, `test_intersect_time`, `test_linear_view`, and others.
Numpy seems not to accept floating point numbers as index anymore.

**Comment by Cadair** [comment]
thanks @olebole I will look into this.

**Comment by Cadair** [comment]
Fixed and backported to 0.7.
